### PR TITLE
Added script and example for zfs

### PR DESCRIPTION
**Pull request by edwardsp**

### CODE REVIEW

File: examples/simple_hpc_pbs_zfs/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+trigger:
+  batch: true
+  branches:
+    include:
+    - master
+    - '*'
+  paths:
+    include:
+      - examples/simple_hpc_pbs_zfs/*
+      - pyazhpc/*
```
Comment (xpillons) [inline examples/simple_hpc_pbs_zfs/pipeline.yml:11]: Please remove that line as we don't want this example to be triggered for every python code change in azhpc

File: examples/simple_hpc_pbs_zfs/readme.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Build a PBS compute cluster
+![Build Status](https://azurecat.visualstudio.com/hpccat/_apis/build/status/azhpc/examples/simple_hpc_pbs?branchName=master)
```
Comment (xpillons) [inline examples/simple_hpc_pbs_zfs/readme.md:2]: This is a wrong link, it should be simple_hpc_pbs_zfs

File: examples/simple_hpc_pbs_zfs/readme.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
+# Build a PBS compute cluster
+![Build Status](https://azurecat.visualstudio.com/hpccat/_apis/build/status/azhpc/examples/simple_hpc_pbs?branchName=master)
+
+Visualisation: [config.json](https://azurehpc.azureedge.net/?o=https://raw.githubusercontent.com/Azure/azurehpc/master/examples/simple_hpc_pbs/config.json)
```
Comment (xpillons) [inline examples/simple_hpc_pbs_zfs/readme.md:4]: please fix the link

File: examples/simple_hpc_pbs_zfs/readme.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+This example will create an HPC cluster ready to run with PBS Pro.  The headnode will export a ZFS filesystem for the nodes to use.
+
+## Initialise the project
+
+To start you need to copy this directory and update the `config.json`.  Azurehpc provides the `azhpc-init` command that can help here by compying the directory and substituting the unset variables.  First run with the `-s` parameter to see which variables need to be set:
+
+```
+azhpc-init -c $azhpc_dir/examples/simple_hpc_pbs -d simple_hpc_pbs -s
```
Comment (xpillons) [inline examples/simple_hpc_pbs_zfs/readme.md:13]: Please fix the dir name as well as below

File: examples/simple_hpc_pbs_zfs/test.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+steps:
```
Comment (xpillons) [inline examples/simple_hpc_pbs_zfs/test.yml:1]: Is there any tests to be done specifically for ZFS that can be added there ?

**Review by xpillons — CHANGES_REQUESTED** [review]

File: examples/simple_hpc_pbs_zfs/pipeline.yml (hunk 1 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/pipeline.yml]: fixed

File: examples/simple_hpc_pbs_zfs/readme.md (hunk 2 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/readme.md]: fixed

File: examples/simple_hpc_pbs_zfs/readme.md (hunk 3 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/readme.md]: fixed

File: examples/simple_hpc_pbs_zfs/readme.md (hunk 4 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/readme.md]: fixed

File: examples/simple_hpc_pbs_zfs/test.yml (hunk 5 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/test.yml:1]: I've added a test to check there is a ZFS pool online

File: examples/simple_hpc_pbs_zfs/readme.md (hunk 2 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/readme.md]: done

File: examples/simple_hpc_pbs_zfs/readme.md (hunk 4 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/readme.md]: done

File: examples/simple_hpc_pbs_zfs/test.yml (hunk 5 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/test.yml:1]: I've added a test to make sure there is at least one ZFS pool online

File: examples/simple_hpc_pbs_zfs/readme.md (hunk 3 above)
Comment (edwardsp) [reply examples/simple_hpc_pbs_zfs/readme.md]: done

**Review by xpillons — APPROVED** [review]
good to me

**Pull request merged**